src/Common/src/TypeSystem/RuntimeDetermined/RuntimeDeterminedType.cs

```diff
@@ -127,7 +127,13 @@ protected override TypeFlags ComputeTypeFlags(TypeFlags mask)
 
         public override TypeDesc GetTypeDefinition()
         {
-            return _rawCanonType.GetTypeDefinition();
+            // TODO: this is needed because NameMangler calls it to see if we're dealing with genericness. Revise?
```

Comment:
Why do we need mangled name for the RuntimeDeterminedType?  I would expect RuntimeDeterminedType to just live at compile time - never materialized in the image.